### ISSUE TITLE
Drop "Button Group" from documentation?

### ISSUE THREAD

**Issue by ffoodd** [Orange-OpenSource/Orange-Boosted-Bootstrap#371]
This pattern is highly questionnable from an accessibility side:
1. may imply a cognitive overload while trying to find a relationship between buttons, even if there is not;
2. may cause error when trying to activate them with mouse, for screen magnifier users or people with vestibular disorders.

I think we should:
- [ ] add spacing between buttons in `.btn-group`, and only remove spacing for specific cases (like toggle button group)
- [ ] remove the "Button Group" page from the doc, to avoid Bootstrap comparison.

**Please note that the current pattern works pretty wel when used with `radio`s or `checkbox`es— it's only missing a `role="group"` or `fieldset`.**

**Comment by ffoodd** [comment]
Wait for [Bootstrap's #30650](https://github.com/twbs/bootstrap/pull/30650) before doing anything here.
